_posts/2025-03-20-phased-strimzi-upgrade-example.md

```diff
@@ -0,0 +1,637 @@
+
+There are several possible ways to implement phased upgrades of the Kafka clusters managed by a Strimzi operator:
+
+### Custom image per Kafka cluster
+
+You can prevent the fleet rolling situation by overriding the [image](https://strimzi.io/docs/operators/latest/configuring#con-common-configuration-images-reference) used by each Kafka cluster. 
+You could manually set this to the version used by the current Strimzi Operator version.
+As you have manually set this, the changes to the image map environment variables when you upgrade the Strimzi Operator would not change the images of the operands.
+So no automatic rolling would occur.
+You could then manually switch the image used by each Kafka cluster to the new Strimzi version. 
+This would allow you to control which clusters rolled and when.
+
+However, you would then enter a situation where a new operator version (for example 0.45) is talking to a Kafka cluster running the old Strimzi version's container images (eg 0.44). 
+As mentioned in the introduction, the Strimzi container images represent an API with certain scripts, endpoints and environment variables expected to be present. 
+Running a newer Strimzi version administering a Kafka cluster using older Strimzi images _could_ lead to issues and is not a recommended or supported deployment model.
+
+### Strimzi per Namespace
+
+Another way to achieve blast radius control is to install a Strimzi Cluster Operator in every namespace where Kafka clusters are running. 
+This way you could upgrade the Cluster Operator Deployment one namespace at a time and have blast radius control (limited to each namespace) and could, if you encounter an issue, roll back a particular Operator upgrade.
+
+However, if you have multiple Kafka clusters in a namespace you would still roll all of them automatically. 
+More seriously, as a single Strimzi instance is capable of running multiple Kafka clusters, this is a very inefficient deployment model with potentially many Strimzi instances running idle. 
+
+The Cluster Operators would also all share the cluster-wide CRDs and so care and time would need to be taken to update these and upgrade all the Operators in a timely fashion.
+
+### Pause Reconciliation
+
+If you have a Strimzi Cluster Operator managing multiple Kafka clusters, either across a whole Kubernetes cluster or in a namespace, then another option is to temporarily stop the Operator from managing a given Kafka cluster (or other operand) until the upgrade is complete.
+
+To do this you would need to:
+1) Apply the [pause reconciliation annotation](https://strimzi.io/docs/operators/latest/deploying#proc-pausing-reconciliation-str) to all Kafka CRs (and other operands like Kafka Connect deployments)
+2) Upgrade the CRDs and the Strimzi Operator Deployment to the new version.
+3) In the order you want, remove the annotation from each Kafka CR. The operator will resume reconciliation, update the image reference and roll the Kafka cluster.
```

Comment:
This does not seem to render. Do you need to use `1.`, `2.` etc.? https://deploy-preview-478--strimzi.netlify.app/blog/2025/03/20/phased-strimzi-upgrade-example/